adapters/tlsio_mbedtls.c

```diff
@@ -1068,17 +1073,15 @@ OPTIONHANDLER_HANDLE tlsio_mbedtls_retrieveoptions(CONCRETE_IO_HANDLE handle)
                 OptionHandler_Destroy(result);
                 result = NULL;
             }
-            else if (&tls_io_instance->owncert != NULL && tls_io_instance->x509_certificate != NULL &&
-                    OptionHandler_AddOption(result, SU_OPTION_X509_CERT, tls_io_instance->x509_certificate) != OPTIONHANDLER_OK)
+            else if (tls_io_instance->x509_certificate != NULL &&
```

Comment:
>tls_io_instance->x509_certificate [](start = 21, length = 33)

is `own_cert` not being used in this case?

Reply:
x509_certificate gets set when own_cert is already set. 

Also checking address of allocated structure (own_cert is memory part of tls_io_instance structure) and comparing it to NULL is always false.


---
In reply to: [356730641](https://github.com/Azure/azure-c-shared-utility/pull/422#discussion_r356730641) [](ancestors = 356730641)

Reply:
ahh i see okay